### DISCUSSION TITLE
Add consistent classNames to elements in sidebar

### DISCUSSION THREAD

**Discussion by evoiron** [storybookjs/storybook#28955] [Ideas]
### Is your feature request related to a problem? Please describe.

Sidebar components are based on emotion, which creates unique classNames. Those classNames may change with each new release. 
It is a problem, when customising sidebar appearence as you may need to update your css on each upgrade.

### Describe the solution you'd like

Each component used in sidebar should have 2 classNames : the one generated by emotion, and another one, which never changes, so that users can use it to customize the sidebar as they like

### Are you able to assist to bring the feature to reality?

no

**Comment by evoiron** [comment]
Anyone to take my request into account ?

**Reply by smlimasu** [reply]
I second this request, but I wanted to ask is there anything specific you might not be able to style atm? I have completely overhauled the appearance of the sidebar without emotion class names, with custom icons, custom margins, hover styles, backgrounds and colors myself using a combination of Id-selectors, properties, pseudo classes and child selectors and element names. With the render label function I inserted custom class names which indicate the depth of the item. 

Some of them are changing from time to time when elements get moved, but I think it's not that frequent  like going after emotion hashed class names. 

So maybe there is also some workaround available for you until semantic class names are there some day?

**Reply by evoiron** [reply]
i missed the render label function. After a quick try, i may be able to do what i want. Thanks for the reply
